crates/sui-indexer-alt/src/models/watermarks.rs

```diff
@@ -85,6 +106,29 @@ impl<'p> CommitterWatermark<'p> {
     }
 }
 
+impl<'p> ReaderWatermark<'p> {
+    pub fn new(pipeline: impl Into<Cow<'p, str>>, reader_lo: u64) -> Self {
+        ReaderWatermark {
+            pipeline: pipeline.into(),
+            reader_lo: reader_lo as i64,
+        }
+    }
+
+    /// Update the reader low watermark for an existing watermark row, as long as this raises the
+    /// watermark, and updates the timestamp this update happened to the database's current time.
+    ///
+    /// Returns a boolean indicating whether the watermark was actually updated or not.
+    pub async fn update(&self, conn: &mut Connection<'_>) -> QueryResult<bool> {
+        Ok(diesel::update(watermarks::table)
+            .set((self, watermarks::pruner_timestamp.eq(diesel::dsl::now)))
+            .filter(watermarks::pipeline.eq(&self.pipeline))
+            .filter(watermarks::reader_lo.lt(self.reader_lo))
+            .execute(conn)
+            .await?
+            > 0)
```

Comment:
i see, so the reader_lo tasks will update on an interval, but whether the writes actually go through occur on the db level, with the filtering conditions here. Neat, and simplifies the task even further